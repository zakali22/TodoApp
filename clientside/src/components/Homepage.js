import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

class Homepage extends Component {
  render() {
    return (
      <div className="homepage">
        <div className="container">
          <div className="homepage__welcome">
            <h1>
              Organize your tasks with simple <span>todos</span>
            </h1>
            <p>
              No more fancy and hard to navigate UI's. <span>Just todo it</span>
              .
            </p>
          </div>
          <div className="homepage__register">
            <Link to="/register">
              <button>Get started</button>
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Homepage;
