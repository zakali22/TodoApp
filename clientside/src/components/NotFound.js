import React from "react";
import PropTypes from "prop-types";

const NotFound = props => {
  return <div>Not found 404</div>;
};

export default NotFound;
