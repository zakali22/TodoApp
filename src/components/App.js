import React, { Component } from "react";
import Form from "./Form";
import List from "./List";
import moment from "moment";

class App extends Component {
  state = {
    todos: [],
    completed: [],
    time: null
  };
  onSubmitHandler = value => {
    const todos = [...this.state.todos];
    const todo = {
      createdAt: moment().calendar(),
      title: value
    };
    this.setState({
      todos: todos.concat(todo)
    });
  };

  onDeleteHandler = id => {
    this.setState({
      todos: this.state.todos.filter((todo, i) => i !== id)
    });
  };

  onDeleteCompleteHandler = id => {
    this.setState({
      completed: this.state.completed.filter((todo, i) => i !== id)
    });
  };

  onEditHandler = (index, value) => {
    const todos = this.state.todos;
    const indexEditAt = this.state.todos.find((todo, i) => i === index);
    const newTodo = {
      ...indexEditAt,
      title: value
    };
    this.setState({
      todos: [
        ...todos.slice(0, index),
        newTodo,
        ...todos.slice(index + 1, todos.length)
      ]
    });
  };

  onCompletionHandler = id => {
    const indexCompleteAt = this.state.todos.find((todo, i) => i === id);
    const completedTodos = [...this.state.completed];
    this.setState({
      todos: this.state.todos.filter((todo, i) => i !== id),
      completed: completedTodos.concat(indexCompleteAt)
    });
  };

  liveClockInterval = () => {
    setInterval(() => {
      this.setState({
        time: moment().format("LTS")
      });
    }, 1000);
  };

  componentDidMount() {
    this.liveClockInterval();
  }

  render() {
    return (
      <div className="main">
        <div className="main__container">
          <div className="main__welcome">
            <h2>Good Morning, Zack</h2>
            <span className="main__welcome--time">
              <p>Its {moment().format("MMMM Do YYYY")}</p>
              <p>{this.state.time}</p>
            </span>
          </div>
          <div className="main__todo">
            <Form onSubmit={this.onSubmitHandler} />
            {this.state.completed.length > 0 ? (
              <List
                todos={this.state.todos}
                completedTodos={this.state.completed}
                onDeleteComplete={this.onDeleteCompleteHandler}
                onDelete={this.onDeleteHandler}
                onEditHandler={this.onEditHandler}
                addToComplete={this.onCompletionHandler}
              />
            ) : (
              <List
                todos={this.state.todos}
                onDelete={this.onDeleteHandler}
                onEditHandler={this.onEditHandler}
                addToComplete={this.onCompletionHandler}
              />
            )}
          </div>
        </div>
      </div>
    );
  }
}

export default App;
