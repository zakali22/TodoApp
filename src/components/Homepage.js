import React, { Component } from "react";
import PropTypes from "prop-types";

class Homepage extends Component {
  render() {
    return (
      <div className="homepage">
        <div className="homepage__welcome">
          <h1>Organize your tasks with simple todos</h1>
          <p>No more fancy and hard to navigate UI's</p>
        </div>
        <div className="homepage__register">
          <button>Get started</button>
        </div>
      </div>
    );
  }
}

export default Homepage;
