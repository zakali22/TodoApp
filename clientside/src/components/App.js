import React, { Fragment } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import PropTypes from "prop-types";
import Homepage from "./Homepage";
import Header from "./Header";
import Todo from "./Todo";
import Signin from "./Signin";
import Register from "./Register";
import NotFound from "./NotFound";
import { connect } from "react-redux";
import PropsRoute from "./routeHelper";

const App = props => {
  return (
    <Router>
      <Fragment>
        <Header />
        <Switch>
          <PropsRoute exact path="/todo" component={Todo} />
          <PropsRoute exact path="/signin" component={Signin} />
          <PropsRoute exact path="/register" component={Register} />
          <PropsRoute exact path="/" component={Homepage} auth={props.auth} />
          <PropsRoute component={NotFound} />
        </Switch>
      </Fragment>
    </Router>
  );
};

const mapStateToProps = state => {
  return {
    auth: state.auth
  };
};

export default connect(
  mapStateToProps,
  null
)(App);
