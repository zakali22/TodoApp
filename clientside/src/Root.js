import React from "react";
import ReactDOM from "react-dom";
import "./index.scss";
import App from "./components/App";

import { createStore, applyMiddleware, compose } from "redux";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import thunk from "redux-thunk";
import rootReducer from "./reducers";
import { Provider } from "react-redux";
import { PersistGate } from "redux-persist/integration/react";

const initialState = {};
const middleware = [thunk];

const persistConfig = {
  key: "root",
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

let store = createStore(
  persistedReducer,
  initialState,
  compose(
    applyMiddleware(...middleware),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  )
);

let persistor = persistStore(store);

export default props => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        {props.children}
      </PersistGate>
    </Provider>
  );
};
