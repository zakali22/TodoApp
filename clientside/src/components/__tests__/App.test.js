import React from "react";
import ReactDOM from "react-dom";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import App from "../App";
import Root from "../../Root";
import Homepage from "../Homepage";
import NotFound from "../NotFound";

describe("App", () => {
  test("has a valid snapshot", () => {
    const component = renderer.create(
      <MemoryRouter>
        <Root>
          <App />
        </Root>
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <Root>
          <App />
        </Root>
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  describe("React Router", () => {
    it("redirects to 404 if invalid path", () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={["/invalidPath"]}>
          <Root>
            <App />
          </Root>
        </MemoryRouter>
      );
      expect(wrapper.find(Homepage)).toHaveLength(0);
      expect(wrapper.find(NotFound)).toHaveLength(1);
    });
    it("doesn't redirect to 404 if path is valid", () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={["/"]}>
          <Root>
            <App />
          </Root>
        </MemoryRouter>
      );
      expect(wrapper.find(Homepage)).toHaveLength(1);
      expect(wrapper.find(NotFound)).toHaveLength(0);
    });
  });
});
