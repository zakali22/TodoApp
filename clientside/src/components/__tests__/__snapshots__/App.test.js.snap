// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div>
  <div
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__nav"
      >
        <a
          className="header__nav--left"
          href="/"
          onClick={[Function]}
        >
          <p>
            Home
          </p>
        </a>
        <div
          className="header__nav--right"
        >
          <a
            className="signup"
            href="/register"
            onClick={[Function]}
          >
            <button>
              Signup
            </button>
          </a>
          <a
            className="login"
            href="/signin"
            onClick={[Function]}
          >
            <button>
              Login
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="homepage"
  >
    <div
      className="container"
    >
      <div
        className="homepage__welcome"
      >
        <h1>
          Organize your tasks with simple 
          <span>
            todos
          </span>
        </h1>
        <p>
          No more fancy and hard to navigate UI's. 
          <span>
            Just todo it
          </span>
          .
        </p>
      </div>
      <div
        className="homepage__register"
      >
        <a
          href="/register"
          onClick={[Function]}
        >
          <button>
            Get started
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
